src/app/aliases/aliases.editor.modal.ts

```diff
@@ -0,0 +1,263 @@
+                                    ie. jamesbond@runbox.com
+                                </mat-hint>
+                                <mat-error *ngFor="let error of field_errors.domain; let i = index;">
+                                    {{error}}
+                                </mat-error>
+                            </div>
+                        </mat-form-field>
+                        <mat-form-field style="margin: 10px; width: 45%;" *ngIf="!is_create">
+                            <input matInput placeholder="Domain"
+                                name="domain"
+                                readonly
+                                [(ngModel)]="data.domain"
+                                (ngModelChange)="onchange_field('domain')"
+                            >
+                        </mat-form-field>
+                        <mat-form-field style="margin: 10px; width: 95%;">
+                            <input matInput placeholder="Deliver to account"
+                                name="forward_to"
+                                [(ngModel)]="data.forward_to"
+                                (ngModelChange)="onchange_field('forward_to')"
+                            >
+                            <mat-hint>
+                                ie. you@runbox.com
+                            </mat-hint>
+                            <div *ngIf="field_errors && field_errors.forward_to">
+                                <mat-hint>
+                                    ie. James Bond
+                                </mat-hint>
+                                <mat-error *ngFor="let error of field_errors.forward_to; let i = index;">
+                                    {{error}}
+                                </mat-error>
+                            </div>
+                        </mat-form-field>
+                    </form>
+                </div>
+          </mat-card-content>
+          <mat-card-actions style="padding: 0px 10px">
+            <button mat-raised-button (click)="save()" color="primary" *ngIf="!is_delete">SAVE</button>
+            <button mat-raised-button (click)="delete()" color="" *ngIf="is_delete">DELETE</button>
+            <button mat-raised-button (click)="close()" color="warn">CANCEL</button>
+          </mat-card-actions>
+          <mat-card-footer>
+          </mat-card-footer>
+        </mat-card>
+    `
+})
+
+export class AliasesEditorModal {
+    @Input() value: any[];
+    field_errors;
+    is_create = false;
+    is_update = false;
+    is_delete = false;
+    has_deleted = false;
+    has_updated = false;
+    has_created = false;
+    allowed_domains = [];
+    constructor(
+        private http: HttpClient,
+        public snackBar: MatSnackBar,
+        public dialog_ref: MatDialogRef<AliasesEditorModal>,
+        @Inject(MAT_DIALOG_DATA) 
+        public data: any,
+        public rmm: RMM,
+    ) {
+        this.load_allowed_domains();
+    }
+    css_class() {
```

Comment:
As above: I think an [Enum](https://www.typescriptlang.org/docs/handbook/enums.html) would do nicely here. Its value could even be the string describing the CSS class, then you won't need an additional lookup table for it:

```
enum EditorMode {
    Create = 'create',
    ...
}
```

Reply:
resolved